### PR TITLE
[Docs] add usefull info to REQUIRE_SIGNIN_VIEW

### PR DESCRIPTION
**Pull request by 6543**
dont happen #9846 again

and apply suggestion of https://github.com/go-gitea/gitea/pull/9784#discussion_r367764502

### CODE REVIEW

File: docs/content/doc/advanced/config-cheat-sheet.en-us.md [hunk 1]
```diff
@@ -309,7 +309,7 @@ relation to port exhaustion.
 - `REQUIRE_EXTERNAL_REGISTRATION_PASSWORD`: **false**: Enable this to force externally created
    accounts (via GitHub, OpenID Connect, etc) to create a password. Warning: enabling this will
    decrease security, so you should only enable it if you know what you're doing.
-- `REQUIRE_SIGNIN_VIEW`: **false**: Enable this to force users to log in to view any page.
+- `REQUIRE_SIGNIN_VIEW`: **false**: Enable this to force users to log in to view any page or API usage.
```
Comment (techknowlogick) [inline docs/content/doc/advanced/config-cheat-sheet.en-us.md]:
```suggestion
- `REQUIRE_SIGNIN_VIEW`: **false**: Enable this to force users to log in to view any page or to use API.
```

File: routers/repo/pull.go [hunk 2]
```diff
@@ -351,7 +351,7 @@ func PrepareViewPullInfo(ctx *context.Context, issue *models.Issue) *git.Compare
 	ctx.Data["Divergence"] = divergence
 	allowUpdate, err := pull_service.IsUserAllowedToUpdate(pull, ctx.User)
 	if err != nil {
-		ctx.ServerError("GetDiverging", err)
+		ctx.ServerError("IsUserAllowedToUpdate", err)
```
Comment (jolheiser) [inline routers/repo/pull.go:354]: I think this was an accidental change?

**Comment by 6543** [comment]
@techknowlogick done

File: routers/repo/pull.go (hunk 2 above)
Comment (jolheiser) [reply routers/repo/pull.go:354]: The change itself makes sense, just not I'm this particular PR.

Comment (6543) [reply routers/repo/pull.go:354]: @jolheiser I know, but this is noice witch I dont like to create a extra PR for one line :/
was from: #9784 ... -> [link](https://github.com/go-gitea/gitea/pull/9784#discussion_r367764502)

Comment (jolheiser) [reply routers/repo/pull.go:354]: Yeah, fair enough. At the very least mention it in the PR description.  
It's never going to matter (probably) but just nice to know it was intentional.

**Review by jolheiser — APPROVED** [review]

**Review by zeripath — APPROVED** [review]

**Comment by 6543** [comment]
Ready to merge :rocker:

**Pull request merged**